rasa/nlu/extractors/entity_synonyms.py

```diff
@@ -44,59 +68,62 @@ def train(
                 entity_val = example.get(TEXT)[entity["start"] : entity["end"]]
                 self.add_entities_if_synonyms(entity_val, str(entity.get("value")))
 
-    def process(self, message: Message, **kwargs: Any) -> None:
+        if self._resource:
+            self._persist()
 
-        updated_entities = message.get(ENTITIES, [])[:]
-        self.replace_synonyms(updated_entities)
-        message.set(ENTITIES, updated_entities, add_to_output=True)
+    def process(self, messages: List[Message]) -> None:
+        """Modifies entities attached to message to resolve synonyms."""
+        for message in messages:
+            updated_entities = message.get(ENTITIES, [])[:]
+            self.replace_synonyms(updated_entities)
+            message.set(ENTITIES, updated_entities, add_to_output=True)
 
-    def persist(self, file_name: Text, model_dir: Text) -> Optional[Dict[Text, Any]]:
+    def _persist(self) -> None:
 
-        if self.synonyms:
-            file_name = file_name + ".json"
-            entity_synonyms_file = os.path.join(model_dir, file_name)
-            write_json_to_file(
-                entity_synonyms_file, self.synonyms, separators=(",", ": ")
-            )
-            return {"file": file_name}
-        else:
-            return {"file": None}
+        if self._synonyms:
+            with self._model_storage.write_to(self._resource) as storage:
+                entity_synonyms_file = os.path.join(
+                    storage, EntitySynonymMapperComponent.SYNONYM_FILENAME
+                )
```

Comment:
`storage` is a `Path` object which allows you to do fancy things like this (see [here](https://docs.python.org/3/library/pathlib.html))
```suggestion
                entity_synonyms_file = storage / EntitySynonymMapperComponent.SYNONYM_FILENAME
                
```